src/libraries/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -177,79 +177,65 @@ public void DirectoryInfo_Create_NullDirectoryInfo()
         {
             DirectoryInfo info = null;
             DirectorySecurity security = new DirectorySecurity();
-
-            Assert.Throws<ArgumentNullException>("directoryInfo", () =>
-            {
-                if (PlatformDetection.IsNetFramework)
-                {
-                    FileSystemAclExtensions.Create(info, security);
-                }
-                else
-                {
-                    info.Create(security);
-                }
-            });
+            Assert.Throws<ArgumentNullException>("directoryInfo", () => DirectoryInfo_Create_Framework(info, security));
         }
 
         [Fact]
         public void DirectoryInfo_Create_NullDirectorySecurity()
         {
             DirectoryInfo info = new DirectoryInfo("path");
-
-            Assert.Throws<ArgumentNullException>("directorySecurity", () =>
-            {
-                if (PlatformDetection.IsNetFramework)
-                {
-                    FileSystemAclExtensions.Create(info, null);
-                }
-                else
-                {
-                    info.Create(null);
-                }
-            });
+            Assert.Throws<ArgumentNullException>("directorySecurity", () => DirectoryInfo_Create_Framework(info, null));
         }
 
         [Fact]
         public void DirectoryInfo_Create_NotFound()
         {
-            using var directory = new TempDirectory();
-            string path = Path.Combine(directory.Path, Guid.NewGuid().ToString(), "ParentDoesNotExist");
-            DirectoryInfo info = new DirectoryInfo(path);
-            DirectorySecurity security = new DirectorySecurity();
+            var directory = new TempAclDirectory();
+            string path = Path.Combine(directory.Path, "ParentDoesNotExist");
+            directory.Dispose(); // Delete parent folder
 
-            Assert.Throws<UnauthorizedAccessException>(() =>
-            {
-                if (PlatformDetection.IsNetFramework)
-                {
-                    FileSystemAclExtensions.Create(info, security);
-                }
-                else
-                {
-                    info.Create(security);
-                }
-            });
+            DirectoryInfo info = new DirectoryInfo(path);
+            DirectorySecurity security = GetDirectorySecurity(FileSystemRights.FullControl);
+            DirectoryInfo_Create_Framework(info, security);
         }
 
-        [Fact]
-        public void DirectoryInfo_Create_DefaultDirectorySecurity()
```

Comment:
I discovered that the unit tests that called the parameter-less `*Security()` constructors, would create undeletable files/folders.

Reply:
This is the error I get when I uncomment the dispose try catch and run this unit test:

```
 System.IO.FileSystemAclExtensionsTests.DirectoryInfo_Create_DefaultDirectorySecurity [FAIL]
        System.UnauthorizedAccessException : Access to the path 'C:\Users\carlos\AppData\Local\Temp\jciofg25.azp\createMe' is denied.
        Stack Trace:
          D:\runtime\src\libraries\System.IO.FileSystem\src\System\IO\Enumeration\FileSystemEnumerator.Win32.cs(87,0): at System.IO.Enumeration.FileSystemEnumerator`1.CreateRelativeDirectoryHandle(ReadOnlySpan`1 relativePath, String fullPath)
          D:\runtime\src\libraries\System.IO.FileSystem\src\System\IO\Enumeration\FileSystemEnumerator.Windows.cs(168,0): at System.IO.Enumeration.FileSystemEnumerator`1.MoveNext()
          D:\runtime\src\libraries\System.IO.FileSystem.AccessControl\tests\TempAclDirectory.cs(24,0): at System.IO.TempAclDirectory.DeleteDirectory()
          D:\runtime\src\libraries\Common\tests\System\IO\TempDirectory.cs(38,0): at System.IO.TempDirectory.Dispose()
          D:\runtime\src\libraries\System.IO.FileSystem.AccessControl\tests\FileSystemAclExtensionsTests.cs(457,0): at System.IO.FileSystemAclExtensionsTests.Verify_DirectorySecurity_CreateDirectory(DirectorySecurity expectedSecurity)
          D:\runtime\src\libraries\System.IO.FileSystem.AccessControl\tests\FileSystemAclExtensionsTests.cs(206,0): at System.IO.FileSystemAclExtensionsTests.DirectoryInfo_Create_DefaultDirectorySecurity()
    Finished:    System.IO.FileSystem.AccessControl.Tests
```

Reply:
@carlossanlop are you also getting this error after fixing https://github.com/dotnet/runtime/pull/47074/files#r568284892 ? We can do some brain storming and find a way to fix it somehow, it would be a pitty to remove a test because the CI does not know how to remove a temp file.

Reply:
I brought it back. I discovered that the problem was in the way I was attempting to reset the permissions in all created files and folders before deleting them. I changed the logic in DeleteDirectory.